indexer/gateway/upload.go

```diff
@@ -0,0 +1,129 @@
+package gateway
+
+import (
+	"encoding/base64"
+	"encoding/json"
+	"net/http"
+	"strings"
+
+	"github.com/0glabs/0g-storage-client/core"
+	"github.com/0glabs/0g-storage-client/core/merkle"
+	"github.com/0glabs/0g-storage-client/node"
+	"github.com/0glabs/0g-storage-client/transfer"
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/gin-gonic/gin"
+	"github.com/pkg/errors"
+)
+
+var expectedReplica uint
+
+type UploadSegmentRequest struct {
+	Root  string          `form:"root" json:"root" binding:"required"`   // File merkle root
```

Comment:
Can we replace the root with txSeq, or add a new field for txSeq here?

In sharding scenario, there may be files with same root but different txSeq, their starting positions on the flow may also be different, which makes the segments that the same node needs to store for these identical files different for each one.